shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/controler/DatasetProcessingApi.java

```diff
@@ -120,4 +122,34 @@ ResponseEntity<Void> updateDatasetProcessing(
 			@Parameter(description = "dataset processing to update", required = true) @Valid @RequestBody DatasetProcessing datasetProcessing, BindingResult result)
 			throws RestServiceException;
 
+	@Operation(summary = "massiveDownloadDatasetsByProcessingIds", description = "If exists, returns a zip file of the inputs/outputs per processing corresponding to the given processing IDs")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "zip file"),
+			@ApiResponse(responseCode = "401", description = "unauthorized"),
+			@ApiResponse(responseCode = "403", description = "forbidden"),
+			@ApiResponse(responseCode = "404", description = "no dataset found"),
+			@ApiResponse(responseCode = "500", description = "unexpected error") })
+	@GetMapping(value = "/massiveDownloadByProcessing")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @datasetSecurityService.HasRightOnEveryDatasetOfProcessings(#processingIds, 'CAN_DOWNLOAD'))")
+	void massiveDownloadByProcessingId(
+			@Parameter(description = "id of the processing", required=true) @Valid
+			@RequestParam(value = "processingIds") List<Long> processingIds,
+			@Parameter(description = "outputs to extract") @Valid
+			@RequestParam(value = "resultOnly") boolean resultOnly, HttpServletResponse response) throws RestServiceException;
+
+	@Operation(summary = "massiveDownloadProcessingDatasetsByExaminationIds", description = "If exists, returns a zip file of the inputs/outputs per processing corresponding to the given examination IDs")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "zip file"),
+			@ApiResponse(responseCode = "401", description = "unauthorized"),
+			@ApiResponse(responseCode = "403", description = "forbidden"),
+			@ApiResponse(responseCode = "404", description = "no dataset found"),
+			@ApiResponse(responseCode = "500", description = "unexpected error") })
+	@GetMapping(value = "/massiveDownloadProcessingByExamination")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @datasetSecurityService.hasRightOnExaminations(#examinationIds, 'CAN_DOWNLOAD'))")
+	void massiveDownloadProcessingByExaminationId(
```

Comment:
same here: 3 names for the same method, in @Operation, @GetMapping and method name, please harmonize to something unique and precise

Reply:
Could you document, what is the expected output of both calls, how does it look like on the disk, once downloaded

Reply:
I added to the description of the @Operation : "Datas are in the http response body, it must be written in a zip file. Datas are sorted with folders according to their respective examination and processing."

It seems enough for me, tell me if it's enough for you